charts/redpanda/templates/post-upgrade.yaml

```diff
@@ -91,7 +91,7 @@ spec:
             if [ -d "/etc/secrets/users/" ]; then
                 IFS=":" read -r USER_NAME PASSWORD MECHANISM < <(grep "" $(find /etc/secrets/users/* -print))
                 curl -svm3 --fail --retry "120" --retry-max-time "120" --retry-all-errors --ssl-reqd \
-  {{- if $cert.caEnabled }}
+  {{- if and $cert ( dig "caEnabled" false $cert ) }}
```

Comment:
That problem repeats in many files

- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-pandaproxy-internal-tls-status.yaml#L65
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-pandaproxy-internal-tls-status.yaml#L74
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L96
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L107
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L123
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L139
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml#L149
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/_configmap.tpl#L600
- https://github.com/redpanda-data/helm-charts/blob/9c4c5854a6deb22558d6dca25ccb485ea2b15eb8/charts/redpanda/templates/_configmap.tpl#L613